pkg/controller/controller.go

```diff
@@ -131,6 +131,11 @@ func (c *Controller) initController() {
 
 	c.logger.Infof("config: %s", c.opConfig.MustMarshal())
 
+	c.checkOperatorServiceAccount(c.config.Namespace)
```

Comment:
As discussed, this shouldn't be here: the service account whose name is specified in `c.opConfig.ServiceAccountName` needs to be present on the watched namespace (so that database pods can use the Kubernetes API), not on the namespace where the operator is deployed.

If the namespace where the operator is deployed does not have a service account named as the one specified in the operator's deployment manifest, then the operator would simply fail to deploy before this code is even running anywhere.